src/nodes.coffee

```diff
@@ -1121,10 +1120,17 @@ exports.Obj = class Obj extends Base
     @lhs = yes
 
     for prop in @properties
+      # Check for reserved words.
+      message = isUnassignable prop.unwrapAll().value
```

Comment:
I think this is already being checked in `Assign` as part of the `eachName` callback (here](https://github.com/GeoffreyBooth/coffeescript/blob/destructuring/src/nodes.coffee#L1752)?